### PR TITLE
Logging, take 2.

### PR DESCRIPTION
**Pull request by mponomar**
Reapply logging + fixes on master.

### CODE REVIEW

File: db/eventlog.c [hunk 1]
```diff
@@ -152,12 +174,58 @@ void eventlog_perfdata(cson_object *obj, const struct reqlogger *logger) {
...
+    return 0;
+}
+
+static void eventlog_context(cson_object *obj, const struct reqlogger *logger) {
+    cson_value *contexts = cson_value_new_array();
+    if (logger->ncontext > 0) {
+        cson_array *arr = cson_value_get_array(contexts);
+        cson_array_reserve(arr, logger->ncontext);
+        for (int i = 0; i < logger->ncontext; i++) {
+            cson_value *v = cson_value_new_string(logger->context[i], strlen(logger->context[i]));
```
Comment (riverszhang89) [inline db/eventlog.c]: Mike would you mind running clang-format on these files? For example this line is definitely more than 80 characters...

**Comment by riverszhang89** [comment]
TODO: port cdb2api context message changes to cdb2jdbc.

**Comment by mponomar** [comment]
Thanks, good catch.  Though we should probably talk about that rule...

**Comment by mponomar** [comment]
Also missing licence...

File: tools/pmux/pmux.cpp [hunk 2]
```diff
@@ -603,6 +629,8 @@ static int run_cmd(struct pollfd &fd, std::vector<struct pollfd> &fds, char *in)
         conn_printf(c, "stat                    : dump some stats\n");
         conn_printf(c, "used                    : dump active port assignments\n");
         conn_printf(c, "rte                     : get route to instance service/port\n");
+        conn_printf(c, "hello service           : keep active connection\n");
```
Comment (riverszhang89) [inline tools/pmux/pmux.cpp]: hello service? hard to associate the command with the usage.

Comment (riverszhang89) [inline tools/pmux/pmux.cpp]: Oh wait. It acts like a HELLO packet.. make sense. Please ignore my comment.

Comment (mponomar) [inline tools/pmux/pmux.cpp]: Keepalive may have been better.

**Comment by mponomar** [comment]
Thanks Rivers.

**Pull request merged**